.github/workflows/ci.yaml

```diff
@@ -0,0 +1,745 @@
+        uses: actions/checkout@v2
+      - name: Restore Python ${{ matrix.python-version }} virtual environment
+        id: cache-venv
+        uses: actions/cache@v2
+        with:
+          path: venv
+          key: >-
+            ${{ runner.os }}-venv-${{ matrix.python-version }}-${{
+            hashFiles('requirements_test.txt') }}-${{
+            hashFiles('requirements_all.txt') }}-${{
+            hashFiles('homeassistant/package_constraints.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-venv-${{ matrix.python-version }}-${{ hashFiles('requirements_test.txt') }}-${{ hashFiles('requirements_all.txt') }}
+            ${{ runner.os }}-venv-${{ matrix.python-version }}-${{ hashFiles('requirements_test.txt') }}
+            ${{ runner.os }}-venv-${{ matrix.python-version }}-
+      - name: Create Python ${{ matrix.python-version }} virtual environment
+        if: steps.cache-venv.outputs.cache-hit != 'true'
+        run: |
+          python -m venv venv
+          . venv/bin/activate
+          pip install -U pip setuptools wheel
+          pip install -r requirements_all.txt -c homeassistant/package_constraints.txt
+          pip install -r requirements_test.txt -c homeassistant/package_constraints.txt
+          # Uninstalling typing as a workaround. Eventually we should make sure
+          # all our dependencies drop typing.
+          # Find offending deps with `pipdeptree -r -p typing`
+          pip uninstall -y typing
+          pip install -e .
+
+  pylint:
+    name: Check pylint
+    runs-on: ubuntu-latest
+    needs: prepare-tests
+    strategy:
+      matrix:
+        python-version: [3.7]
+    container: homeassistant/ci-azure:${{ matrix.python-version }}
+    steps:
+      - name: Check out code from GitHub
+        uses: actions/checkout@v2
+      - name: Restore Python ${{ matrix.python-version }} virtual environment
+        id: cache-venv
+        uses: actions/cache@v2
+        with:
+          path: venv
+          key: >-
+            ${{ runner.os }}-venv-${{ matrix.python-version }}-${{
+            hashFiles('requirements_test.txt') }}-${{
+            hashFiles('requirements_all.txt') }}-${{
+            hashFiles('homeassistant/package_constraints.txt') }}
+      - name: Fail job if Python cache restore failed
+        if: steps.cache-venv.outputs.cache-hit != 'true'
+        run: |
+          echo "Failed to restore Python virtual environment from cache"
+          exit 1
+      - name: Run pylint
+        run: |
+          . venv/bin/activate
+          pylint homeassistant
+
+  mypy:
+    name: Check mypy
+    runs-on: ubuntu-latest
+    needs: prepare-tests
+    strategy:
+      matrix:
+        python-version: [3.7]
+    container: homeassistant/ci-azure:${{ matrix.python-version }}
+    steps:
+      - name: Check out code from GitHub
+        uses: actions/checkout@v2
+      - name: Restore Python ${{ matrix.python-version }} virtual environment
```

Comment:
Could we indicate in the name if it is using the `requirements_all` file or not? Maybe use terms `slim`/`full`

Reply:
Done, used `base` & `full`. As the base term was already used 😉